core/icon_api.php

```diff
@@ -34,6 +34,42 @@
 require_api( 'helper_api.php' );
 require_api( 'utility_api.php' );
 
+/**
+ * Returns HTML to display an icon with optional title pop-up.
+ *
+ * @param string $p_icon       Fontawesome icon name (fa-xxx)
+ * @param string $p_classes    Optional additional CSS classes
+ * @param string $p_title      Optional pop-up text
+ * @param string $p_inner_html Optional html to insert with the icon.
+ *
+ * @return string
+ */
+function icon_get( $p_icon, $p_classes = '', $p_title = '', $p_inner_html = '' ) {
+	# Add 'fa-' prefix if missing
+	if( substr( $p_icon, 0, 3 ) != 'fa-' ) {
```

Comment:
While this "repairs" bugs in some way, it hides them and adds runtime overhead.
I would remove it.

Reply:
I don't consider this as "repairing a bug".

FontAwesome icons are defined with a name, which is what the API expects. The `fa-` prefix is just a CSS convention, and my idea was to provide an API to deal with _icons_, not CSS classes. So this piece of code is here to provide backwards compatibility.

That said, I hear the argument about overhead, although that is quite negligible IMO. Let's hear what other devs think.

Reply:
@vboctor your feedback on this would be appreciated.